src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\SimpleThingsEntityAudit\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use SimpleThings\EntityAudit\AuditManager;
+
+/**
+ * Creates a resource metadata from EntityAudit Entity.
+ *
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+final class EntityAuditResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $resourceMetadataFactory;
+    private $auditManager;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory = null, AuditManager $auditManager)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->auditManager = $auditManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass) : ResourceMetadata
+    {
+        $resourceClassMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+        if (true === $this->auditManager->getMetadataFactory()->isAudited($resourceClass)) {
+            $resourceAuditedMetadata = new ResourceMetadata(
+                    $resourceClassMetadata->getShortName(),
```

Comment:
I feel like it is more readable like this, and there is no problem with php-cs / styleci, what is the problem for you in here ? 🐰